### ISSUE TITLE
API not working

### ISSUE THREAD

**Issue by aldareth** [k3min/infinigag#24]
Hello,

Since 1 or 2 days the API is not working anymore. The GET section method or the GET post method are not loading results for example, it keeps loading indefinitely.

Can you do something to fix that ? Thanks

Regards

**Comment by ghost** [comment]
Same here, the API is not working anymore... :( !

**Comment by ghost** [comment]
Same here

**Comment by EloiStree** [comment]
When you spend sometime on a Plug-In and then you put your code to the trash because the API is not working anymore :-1: I suppose I will have to do my own hacking system with Selenium to access those data.

**Comment by riddhimanadib** [comment]
The APIs are still not working. Please let us know if the issue is resolved!

**Comment by tiagomartinho** [comment]
Yes, it is not working, I also had to remove an app that I had in the app store.

Does anyone know an alternative?

**Comment by EloiStree** [comment]
The alternative is to listen to an RSS server, for example:
http://9gag-rss.com/
But it is the same problem. 
If the server is down you are screw.

Or you can learn about Selenium:
http://www.seleniumhq.org/
Let's a virtual machine turn all time to create your own API.
This way when there is a bug, you can easily changed it in 1-3 hours ;)
The down side is that you are directly affected by the 9gag website modification.

**Comment by tiagomartinho** [comment]
@JamsCenter thank you very much for the hint, I think I will use the RSS-Feed a custom solution does not fit my needs right now.

**Comment by ghost** [comment]
Same here :(, Two apps on appstore are not working, hope you get it back soon.
RSS only seems to return 6 items so not really a solution for me.

**Comment by EloiStree** [comment]
Hope the owner do not have problem in life.
Because he is not responding at all.

**Comment by ymerdrengene** [comment]
Not working for me either. Please fix this

**Comment by ghost** [comment]
RIP InfiniGag...
Sad story, it was a nice tool.

**Comment by GGsP839X** [comment]
Some months ago, I contacted k3min and asked him about the ability to post something to 9GAG. He responded some weeks later that he is very busy, so I assume he still is.

Anyway, there is no (easy) possibility to access 9GAG without a complete virtual 'near-browser-solution', because you need javascript to access the content, which is loaded part by part after the main site container. And that's where things get complicated.

It might be a solution to use [PhantomJS](https://github.com/ariya/phantomjs) (also [available as a cloud service](https://phantomjscloud.com/index.html) if you do not have the resources to run it on your own server) to capture the javascript loaded content correctly, but 9GAG pulls content every two seconds, so I think this might work only partially. Someone should give it a try.

PS: If you want to take a look at an [attempt to do it with PHP and cURL](http://stackoverflow.com/questions/37173719/php-curl-login-with-csrf-token-not-working-9gag-com) (which failed because of the reason named above).

**Comment by EloiStree** [comment]
There is an easy way:
http://docs.seleniumhq.org/

It is because I do not have the time to do it.
But I did it for hacking Ogame.
It works like a charme ;)

**Comment by GGsP839X** [comment]
The problem with Selenium is that you cannot run it on a server as easy as PhantomJS and that it gets quite CPU consuming on multiple instances (which you will need for many clients accessing simultaneously).

**EDIT:**
So, five hours later now, **I managed to access 9GAG and to login successfully** through a normal webhoster using PHP. Although I have not found a way to post comments yet (9GAGs structure is quite challenging), I am able to fetch posts, view images and so on, so basically all the stuff you could do with 9GAG.
I do not know if I will develop this any further, but if anyone is interested in developing a (new) API, I will share my solution with you.

**Comment by austinhuang0131** [comment]
RIP API. I moved to [node-9gag](https://www.npmjs.com/package/node-9gag) now.
Oh, maybe you should try host an API with that? It's node.js so you'll be _easier_ to make another JSON api.

**Comment by kingalandydy** [comment]
Yeah this sucks. I built something with this and it doesn't work anymore. Why? :(

**Comment by divinity76** [comment]
just a note, 

Selenium is a ram & cpu -intensive full-fledged browser.

libcurl could do all this api does, at a fraction of the system resources Selenium requires. you probably should look into libcurl instead

**Comment by GGsP839X** [comment]
Are you interested in developing a new API?
I can share my PHP solution with you if you are interested!

**Comment by kingalandydy** [comment]
Yes please share

**Comment by eeshankeni** [comment]
RIP
